core/src/main/scala/Main.scala

```diff
@@ -1,21 +1,19 @@
 import zio.{Console, Scope, ZIO, ZIOAppDefault, ZLayer}
 
 object Main extends ZIOAppDefault:
+  val program: ZIO[CarApi, Nothing, Unit] =
+    for {
+      api <- ZIO.service[CarApi]
+      _ <- api.register("Toyota Corolla WE98765").debug
+      _ <- api.register("VW Golf WN12345").debug
+      _ <- api.register("Tesla").debug
+    } yield ()
+
   override def run: ZIO[Any, Any, Any] =
-    ZIO.scoped {
-      ZLayer
-        .makeSome[Scope, CarApi](
-          CarApi.live,
-          CarService.live,
-          CarRepository.live,
-          DB.live,
-          ConnectionPool.live
-        )
-        .build
-        .map(_.get[CarApi])
-        .flatMap { api =>
-          api.register("Toyota Corolla WE98765").flatMap(Console.printLine(_)) *>
-            api.register("VW Golf WN12345").flatMap(Console.printLine(_)) *>
-            api.register("Tesla").flatMap(Console.printLine(_))
-        }
-    }
+    program.provide(
```

Comment:
So here `.provide` will allocate the layer, use it for the given effect, and then release it - releasing any resources that were acquired during the construction phase? It wasn't quite clear what the lifecycle of the layers is looking at the scaladoc and which methods of using a layer trigger which lifecycle events